engine/objconv/mesher/to_BFXM.cpp

```diff
@@ -481,13 +481,19 @@ void beginElement( const string &name, const AttributeList &attributes, XML *xml
     case XML::POLYGONS:
         break;
     case XML::LINE:
-        memset( &xml->linetemp, 0, sizeof (xml->linetemp) );
+        xml->linetemp.flatshade = 0;
+        memset(xml->linetemp.indexref, 0, sizeof(xml->linetemp.indexref) );
+        memset(xml->linetemp.s, 0, sizeof(xml->linetemp.s));
+        memset(xml->linetemp.t, 0, sizeof(xml->linetemp.t));
```

Comment:
So, when doing `sizeof` it's better to use native types:
```
sizeof(float) * 5
sizeof(char)* 10
```
Than do use `sizeof(usertype)`. Reason being, that the variable being referenced (usertype) might be a pointer and will then give the wrong value. That is...

The intent is to clear the data (int :arrow_right: 0), but the parameter to `sizeof` is a pointer, so it calculates the size of a pointer (32-bit, 64-bit) and uses that instead.

For instance, on a 64-bit system:
```
sizeof(uint32) = 4 // (4 byte data type)
sizeof(*uint32) = 8 // (8 byte pointer)
```
Some times that happens to match; for instance the same code on a 32-bit system:
```
sizeof(uint32) = 4 // (4 byte data type)
sizeof(*uint32) = 4 // (4 byte pointer)
```

So I find it's always best to use built-in types to `sizeof` instead - explicit being better than implicit.

For this, code above, it'd be better to do:
```
memset(xml-linetemp.indexref, 0, sizeof(struct line))
```

Actually...you might want to just add a `clear()` on `struct line` insetad of using `memset` directly...probably similar in the other items (triangle, quad, etc) in `to_BFXM.h` too.

Reply:
For memset(xml->linetemp.indexref, 0, sizeof(struct line)); would that not clear a block of memory the size of the entire struct? I just want to clear the array in this case. 
I've gone with your initial suggestion and specified the type and number at first, but then I searched stackoverload.

Because that clear()  was a little bit of a mystery, as there are a few libraries that provide different versions of it. In short, I did not know how to use it. 
Turns out, in C11, you can just tell the compiler with a plain old "struct = {};" that you want the structure to be empty.  I had a look at the structs, and they are properly defined to allow this sort of thing. Learn something every day!

Reply:
I'm not familiar with the `{}` method. I'll have to look that up before I approve.

Sorry if I wasn't clear enough, the `clear` suggestion was to add a function on the struct to do the work that would just get called.

Reply:
@BenjamenMeyer , Ah, I see. Yes, that could work, too. If you are not happy with the = {} method, I'll implement that. As far as I know, the = {} method uses automatically generated destructors and constructors. So, it's implicit code, rather than explicit, and you have already expressed a preference for explicit code....

Reply:
@Benjamin, I have made the clear routine in the structure, and used it in this routine.